apps/shell/tash_main.c

```diff
@@ -151,19 +151,22 @@ char *tash_read_input_line(int fd)
 #if !defined(CONFIG_DISABLE_POLL)
 		FD_ZERO(&tfd);
 		FD_SET(fd, &tfd);
-
+#ifdef CONFIG_PM
+		tash_pm_resume();
+#endif
 		if ((select(fd + 1, &tfd, NULL, NULL, &stimeout)) && FD_ISSET(fd, &tfd)) {
```

Comment:
how about seperate select?
we can call pm_suspend only once location
```suggestion
		ret = select(fd + 1, &tfd, NULL, NULL, &stimeout);
		tash_pm_suspend();
		if (ret > 0 && FD_ISSET(fd, &tfd)) {
		     tash_pm_timedsuspend(TASH_PM_TIMEDSUSPEND_TIME_IN_MS);

```